coffea/processor/executor.py

```diff
@@ -1073,157 +817,450 @@ def run_uproot_job(
+                item.filename,
+                timeout=self.xrootdtimeout,
+                file_handler=uproot.MemmapSource
+                if self.mmap
+                else uproot.MultithreadedFileSource,
+            )
+        elif self.format == "parquet":
+            filecontext = ParquetFileContext(item.filename)
+
+        metadata = {
+            "dataset": item.dataset,
+            "filename": item.filename,
+            "treename": item.treename,
+            "entrystart": item.entrystart,
+            "entrystop": item.entrystop,
+            "fileuuid": str(uuid.UUID(bytes=item.fileuuid))
+            if len(item.fileuuid) > 0
+            else "",
+        }
+        if item.usermeta is not None:
+            metadata.update(item.usermeta)
+
+        with filecontext as file:
+            if self.schema is None:
+                # To deprecate
+                tree = file[item.treename]
+                events = LazyDataFrame(
+                    tree, item.entrystart, item.entrystop, metadata=metadata
+                )
+            elif issubclass(self.schema, schemas.BaseSchema):
+                # change here
+                if self.format == "root":
+                    materialized = []
+                    factory = NanoEventsFactory.from_root(
+                        file=file,
+                        treepath=item.treename,
+                        entry_start=item.entrystart,
+                        entry_stop=item.entrystop,
+                        persistent_cache=self.cache,
+                        schemaclass=self.schema,
+                        metadata=metadata,
+                        access_log=materialized,
+                    )
+                    events = factory.events()
+                elif self.format == "parquet":
+                    rados_parquet_options = {}
+                    if ":" in item.filename:
+                        ceph_config_path, filename = item.filename.split(":")
+                        # patch back filename into item
+                        item = WorkItem(**dict(asdict(item), filename=filename))
+                        rados_parquet_options["ceph_config_path"] = ceph_config_path
+
+                    factory = NanoEventsFactory.from_parquet(
+                        file=item.filename,
+                        treepath=item.treename,
+                        metadata=metadata,
+                        rados_parquet_options=rados_parquet_options,
+                    )
+                    events = factory.events()
+            else:
+                raise ValueError(
+                    "Expected schema to derive from nanoevents.BaseSchema, instead got %r"
+                    % self.schema
+                )
+            tic = time.time()
+            try:
+                out = processor_instance.process(events)
+            except Exception as e:
+                file_trace = f"\n\nFailed processing file: {item.filename} ({item.entrystart}-{item.entrystop})"
```

Comment:
Now that `WorkItem` is a fancy dataclass, I think its repr can be used here

Reply:
done